### ISSUE TITLE
ITMS-90428: Invalid Swift Support 

### ISSUE THREAD

**Issue by idanasher** [airsdk/Adobe-Runtime-Support#892]
After completing & compiling my distribution ipa , I uploaded it via the Transporter app on my mac.

After the upload, I got this message from the app store connect:

**ITMS-90428: Invalid Swift Support** 
- The files libswiftFoundation.dylib, libswiftos.dylib, libswiftMetal.dylib, libswiftDispatch.dylib, libswiftUIKit.dylib, libswiftObjectiveC.dylib, libswiftCoreGraphics.dylib, libswiftDarwin.dylib, libswiftCoreFoundation.dylib, libswiftCore.dylib, libswiftCoreImage.dylib, libswiftQuartzCore.dylib don’t match /Payload/Rummy World.app/Frameworks/libswiftFoundation.dylib, /Payload/Rummy World.app/Frameworks/libswiftos.dylib, /Payload/Rummy World.app/Frameworks/libswiftMetal.dylib, /Payload/Rummy World.app/Frameworks/libswiftDispatch.dylib, /Payload/Rummy World.app/Frameworks/libswiftUIKit.dylib, /Payload/Rummy World.app/Frameworks/libswiftObjectiveC.dylib, /Payload/Rummy World.app/Frameworks/libswiftCoreGraphics.dylib, /Payload/Rummy World.app/Frameworks/libswiftDarwin.dylib, /Payload/Rummy World.app/Frameworks/libswiftCoreFoundation.dylib, /Payload/Rummy World.app/Frameworks/libswiftCore.dylib, /Payload/Rummy World.app/Frameworks/libswiftCoreImage.dylib, /Payload/Rummy World.app/Frameworks/libswiftQuartzCore.dylib. Make sure the files are correct, rebuild your app, and resubmit it. Don’t apply post-processing to /Payload/Rummy World.app/Frameworks/libswiftFoundation.dylib, /Payload/Rummy World.app/Frameworks/libswiftos.dylib, /Payload/Rummy World.app/Frameworks/libswiftMetal.dylib, /Payload/Rummy World.app/Frameworks/libswiftDispatch.dylib, /Payload/Rummy World.app/Frameworks/libswiftUIKit.dylib, /Payload/Rummy World.app/Frameworks/libswiftObjectiveC.dylib, /Payload/Rummy World.app/Frameworks/libswiftCoreGraphics.dylib, /Payload/Rummy World.app/Frameworks/libswiftDarwin.dylib, /Payload/Rummy World.app/Frameworks/libswiftCoreFoundation.dylib, /Payload/Rummy World.app/Frameworks/libswiftCore.dylib, /Payload/Rummy World.app/Frameworks/libswiftCoreImage.dylib, /Payload/Rummy World.app/Frameworks/libswiftQuartzCore.dylib.

AIR 33.1.1.476 fro ios
Compiling with Animate 2020 on Windows 10

**Comment by spielepalast** [comment]
What is the status of this ticket?

**Comment by idanasher** [comment]
@spielepalast , the status is "in process"

@ajwfrost  Andrew is working on the air sdk and trying to solve this issue.

You can see this tread as well for updates:
https://github.com/airsdk/Adobe-Runtime-Support/issues/776

**Comment by ajwfrost** [comment]
Should be fixed in 33.1.1.533, please check and re-open if necessary.
Thanks

**Comment by urthling** [comment]
Just pushed a 533 build to the App Store and I'm seeing the same:

ITMS-90428: Invalid Swift Support

 - The files libswiftObjectiveC.dylib, libswiftCoreGraphics.dylib, libswiftUIKit.dylib, libswiftos.dylib, libswiftMetal.dylib, libswiftDispatch.dylib, libswiftCoreImage.dylib, libswiftCore.dylib, libswiftFoundation.dylib, libswiftQuartzCore.dylib, libswiftDarwin.dylib, libswiftCoreFoundation.dylib don’t match /Payload/Scuba.app/Frameworks/libswiftObjectiveC.dylib, /Payload/Scuba.app/Frameworks/libswiftCoreGraphics.dylib, /Payload/Scuba.app/Frameworks/libswiftUIKit.dylib, /Payload/Scuba.app/Frameworks/libswiftos.dylib, /Payload/Scuba.app/Frameworks/libswiftMetal.dylib, /Payload/Scuba.app/Frameworks/libswiftDispatch.dylib, /Payload/Scuba.app/Frameworks/libswiftCoreImage.dylib, /Payload/Scuba.app/Frameworks/libswiftCore.dylib, /Payload/Scuba.app/Frameworks/libswiftFoundation.dylib, /Payload/Scuba.app/Frameworks/libswiftQuartzCore.dylib, /Payload/Scuba.app/Frameworks/libswiftDarwin.dylib, /Payload/Scuba.app/Frameworks/libswiftCoreFoundation.dylib. Make sure the files are correct, rebuild your app, and resubmit it. Don’t apply post-processing to /Payload/Scuba.app/Frameworks/libswiftObjectiveC.dylib, /Payload/Scuba.app/Frameworks/libswiftCoreGraphics.dylib, /Payload/Scuba.app/Frameworks/libswiftUIKit.dylib, /Payload/Scuba.app/Frameworks/libswiftos.dylib, /Payload/Scuba.app/Frameworks/libswiftMetal.dylib, /Payload/Scuba.app/Frameworks/libswiftDispatch.dylib, /Payload/Scuba.app/Frameworks/libswiftCoreImage.dylib, /Payload/Scuba.app/Frameworks/libswiftCore.dylib, /Payload/Scuba.app/Frameworks/libswiftFoundation.dylib, /Payload/Scuba.app/Frameworks/libswiftQuartzCore.dylib, /Payload/Scuba.app/Frameworks/libswiftDarwin.dylib, /Payload/Scuba.app/Frameworks/libswiftCoreFoundation.dylib

**Comment by ajwfrost** [comment]
Can I check what build target type you used - was it "ipa-app-store"?
Any chance we can look at your IPA file please? https://transfer.harman.com/requests/TQ1vfER8YcDkRh5jp7nLUr
thanks

**Comment by urthling** [comment]
Aye, using: `-target ipa-app-store`  -- just sent the IPA..

**Comment by ajwfrost** [comment]
Thanks. So that all looks fine, the signatures check out, and apart from these signatures the content of each architecture block are identical. From what we could tell - having inspected an IPA that previously worked - this is all good.

But, the Apple message suggests that there should be no post-processing done on these other dylib files. The two differences that we do have are (a) we're only including the architectures that we're using, and (b) we're re-signing them.

Let me see whether we can change either/both of those, and get it to a point where Apple accept it...

**Comment by urthling** [comment]
Gotcha, sounds good..  Just fyi, I did notice it's about 20MB larger than a .476 build..

**Comment by ajwfrost** [comment]
Yes, the size difference is likely to be because we are just moving the swift libraries into the "SwiftSupport" folder unchanged .. which means they have all 4 CPU architecture variants in them.

It would be lovely if Apple published details of what they actually want this IPA format to contain.. sadly it's a bit of trial and error, and inspecting packages that work, for us to work this out...

**Comment by urthling** [comment]
Understood..  power to you and the team!

**Comment by ajwfrost** [comment]
Hi @urthling  - we have a version of ADT that's got some extra configuration settings in it, are you able to try this please?
Details are in the below message:
https://transfer.harman.com/message/bOpJ9PNdRCeyCRsJQ6mgQ9
Essentially, if you can use this and add the below two lines to the "lib\adt.cfg" file, it may then end up being accepted by Apple:
```
OnlyIncludeSwiftUsedArchsInSupport=false
OnlyIncludeSwiftUsedArchsInPayload=false
```
thanks

**Comment by urthling** [comment]
You bet -- will report back shortly..

**Comment by ajwfrost** [comment]
Great, thanks for checking that! So I think we should set this behaviour as 'default' for app store release builds..

**Comment by urthling** [comment]
Quick note on this -- I notice that the 554 adt.cfg does not have the two entries

```
OnlyIncludeSwiftUsedArchsInSupport=false
OnlyIncludeSwiftUsedArchsInPayload=false
```

I'm getting the `ITMS-90428: Invalid Swift Support` email from Apple

After adding them back in -- I'm still getting this email from Apple

**Comment by ajwfrost** [comment]
Yes, those changes came after the 554 release candidate was built. We're going to have to do (yet) another release to update the behaviour here so that these values are default false...
In the meantime, the ADT from the above link was I think built on top of the .554 release candidate anyway so you can use that for the meantime.

thanks

**Comment by urthling** [comment]
Ah, my bad, gotcha, k, thank you Andrew
